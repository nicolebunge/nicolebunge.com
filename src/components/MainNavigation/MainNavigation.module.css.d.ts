/* eslint-disable camelcase */

declare const styles: {
  readonly mainNavigation: string;
  readonly mainNavigationIsActive: string;
  readonly mainNavigation__button: string;
  readonly mainNavigation__list: string;
  readonly mainNavigation__listInline: string;
  readonly mainNavigation__item: string;
  readonly mainNavigation__link: string;
  readonly mainNavigation__linkActive: string;
};
export = styles;
