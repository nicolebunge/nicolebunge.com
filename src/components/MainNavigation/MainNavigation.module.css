.mainNavigation {
  font-weight: 600;
}

.mainNavigation--is-active {
  background-color: #fff;
  bottom: 0;
  display: grid;
  height: 100vh;
  left: 0;
  place-content: center;
  position: fixed;
  right: 0;
  text-align: center;
  top: 0;
  width: 100vw;
  z-index: 1;
}

.mainNavigation--is-active .mainNavigation__button {
  position: absolute;
  right: 5%;
  top: 0.5em;
}

.mainNavigation__list {
  display: none;
  flex-wrap: wrap;
  list-style: none;
  margin: -1em;
  padding: 0;
}

.mainNavigation--is-active .mainNavigation__list {
  display: block;
  margin: 0;
}

.mainNavigation__link {
  display: inline-block;
  padding-bottom: var(--space-lg);
  padding-top: var(--space-lg);
  position: relative;
  transition: color 0.3s;
}

.mainNavigation__link--active {
  color: var(--color-primary);
}

@media (min-width: 640px) {
  .mainNavigation__button {
    display: none;
  }

  .mainNavigation__item {
    margin: 1em;
  }

  .mainNavigation__list {
    display: flex;
  }

  .mainNavigation__link--active::after {
    background-color: currentcolor;
    content: "";
    height: 0.25em;
    position: absolute;
    right: 0;
    top: 0;
    width: 100%;
  }
}
