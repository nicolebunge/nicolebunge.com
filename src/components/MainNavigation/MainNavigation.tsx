/* eslint-disable camelcase */

import clsx from 'clsx';
import React, { HTMLAttributes } from 'react';
import { useLockBodyScroll, useToggle } from 'react-use';
import Link from '../Link';
import {
  mainNavigation,
  mainNavigationIsActive,
  mainNavigation__button,
  mainNavigation__item,
  mainNavigation__link,
  mainNavigation__linkActive,
  mainNavigation__list,
  mainNavigation__listInline,
} from './MainNavigation.module.css';
import { Menu1, XMark1 } from '../Icons';

interface Route {
  name: string;
  path: string;
}

export interface NavigationProps extends HTMLAttributes<HTMLDivElement> {
  inline?: boolean;
  routes: Route[];
}

function MainNavigation(props: NavigationProps): JSX.Element {
  const { className, inline = true, routes, ...otherProps } = props;
  const [isActive, toggle] = useToggle(false);

  useLockBodyScroll(isActive);

  return (
    <nav
      className={clsx(className, mainNavigation, {
        [mainNavigationIsActive]: isActive,
      })}
      {...otherProps}
    >
      <button className={mainNavigation__button} type="button" onClick={toggle}>
        {isActive ? <XMark1 /> : <Menu1 />}
      </button>

      {routes && (
        <ul
          className={clsx(mainNavigation__list, {
            [mainNavigation__listInline]: inline,
          })}
        >
          {routes.map((route) => (
            <li className={mainNavigation__item} key={route.path}>
              <Link
                to={route.path}
                className={mainNavigation__link}
                activeClassName={mainNavigation__linkActive}
              >
                {route.name}
              </Link>
            </li>
          ))}
        </ul>
      )}
    </nav>
  );
}

export default MainNavigation;
