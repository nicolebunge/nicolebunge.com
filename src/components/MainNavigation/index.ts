import MainNavigation from './MainNavigation';

export * from './MainNavigation';
export default MainNavigation;
