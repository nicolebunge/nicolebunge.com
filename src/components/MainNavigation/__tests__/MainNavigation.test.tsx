import { render } from '@testing-library/react';
import React from 'react';
import { axe } from 'jest-axe';
import MainNavigation from '..';

describe('MainNavigation component', () => {
  it('renders correctly', async () => {
    const routes = [
      {
        name: 'Index',
        path: '/',
      },
    ];
    const { container } = render(<MainNavigation routes={routes} />);

    expect(container.firstChild).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
