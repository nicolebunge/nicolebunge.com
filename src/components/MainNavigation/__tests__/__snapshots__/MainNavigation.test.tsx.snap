// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation component renders correctly 1`] = `
<nav
  aria-label="Hauptmenü"
  class="mainNavigation"
  role="navigation"
>
  <button
    aria-label="Menü öffnen"
    class="mainNavigation__button"
    type="button"
  >
    <svg
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M24 6H0V2h24v4zm0 4H0v4h24v-4zm0 8H0v4h24v-4z"
      />
    </svg>
  </button>
  <ul
    class="mainNavigation__list"
  >
    <li
      class="mainNavigation__item"
    >
      <a
        class="mainNavigation__link"
        href="/"
      >
        Index
      </a>
    </li>
  </ul>
</nav>
`;
